docs/examples/101/function-http-trigger/main.bicep

```diff
@@ -0,0 +1,152 @@
+        }
+        {
+          name: 'WEBSITE_CONTENTAZUREFILECONNECTIONSTRING'
+          value: 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};EndpointSuffix=${environment().suffixes.storage};AccountKey=${listKeys(storageAccount.id, storageAccount.apiVersion).keys[0].value}'
+        }
+        {
+          name: 'APPINSIGHTS_INSTRUMENTATIONKEY'
+          value: appInsights.properties.InstrumentationKey
+        }
+        {
+          name: 'APPLICATIONINSIGHTS_CONNECTION_STRING'
+          value: 'InstrumentationKey=${appInsights.properties.InstrumentationKey}'
+        }
+        {
+          name: 'FUNCTIONS_WORKER_RUNTIME'
+          value: functionRuntime
+        }
+        {
+          name: 'FUNCTIONS_EXTENSION_VERSION'
+          value: '~3'
+        }
+      ]
+    }
+    httpsOnly: true
+  }
+}
+
+resource function 'Microsoft.Web/sites/functions@2020-06-01' = {
+  name: '${functionApp.name}/${functionNameComputed}'
+  properties: {
+    config: {
+      disabled: false
+      bindings: [
+        {
+          name: 'req'
+          type: 'httpTrigger'
+          direction: 'in'
+          authLevel: 'function'
+          methods: [
+            'get'
+          ]
+        }
+        {
+          name: '$return'
+          type: 'http'
+          direction: 'out'
+        }
+      ]
+    }
+    files: {
+      'run.csx': '#r "Newtonsoft.Json"\n\nusing System.Net;\nusing Microsoft.AspNetCore.Mvc;\nusing Microsoft.Extensions.Primitives;\nusing Newtonsoft.Json;\n\npublic static async Task<IActionResult> Run(HttpRequest req, ILogger log)\n{\n    log.LogInformation("C# HTTP trigger function processed a request.");\n\n    string name = req.Query["name"];\n\n    string requestBody = await new StreamReader(req.Body).ReadToEndAsync();\n    dynamic data = JsonConvert.DeserializeObject(requestBody);\n    name = name ?? data?.name;\n\n    string responseMessage = string.IsNullOrEmpty(name)\n        ? "This HTTP triggered function executed successfully. Pass a name in the query string or in the request body for a personalized response."\n                : $"Hello, {name}. This HTTP triggered function executed successfully.";\n\n            return new OkObjectResult(responseMessage);\n}\n'
+    }
+  }
+}
+
+resource keyVault 'Microsoft.KeyVault/vaults@2019-09-01' = {
+  name: keyVaultName
+  location: location
+  properties: {
+    tenantId: subscription().tenantId
+    sku: {
+      family: 'A'
+      name: keyVaultSku
+    }
+    accessPolicies: []
+  }
+}
+
+resource keyVaultSecret 'Microsoft.KeyVault/vaults/secrets@2018-02-14' = {
+  name: '${keyVault.name}/${functionAppKeySecretName}'
+  properties: {
+    value: listKeys('${resourceId('Microsoft.Web/sites', functionApp.name)}/host/default', '2020-06-01').functionKeys.default

```

Comment:
nit - but assuming this default site resource has the same apiVersion as your farm and/or function site, you can probably replace the hardcoded apiVersion with `functionApp.apiVersion` or `function.apiVersion`

Reply:
Just to add, I think you should be able to simplify further using the `.id` value e.g.:
```
value: listKeys('${functionApp.id}/host/default', functionApp.apiVersion).functionKeys.default
```

Reply:
Thanks @alex-frankel and @anthony-c-martin - I have made those changes.